app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java

```diff
@@ -46,6 +47,10 @@
     public void onCreate() {
         xdrip.context = getApplicationContext();
         super.onCreate();
+
+        //start widget update service to capture time ticks
+        this.startService(new Intent(this, WidgetUpdateService.class));
```

Comment:
This is a problem because its going to enable the service even if no widgets are active.  How about I make a new preference option in `Smart Watch Features` for Galaxy Gear / Wearable Widgets with an option users can enable. We can then make this early start conditional on that?

Reply:
I don't think we need to add another setting for this. The existence of Gear widgets can be determined dynamically, so the service (and the clock tick event) can be handled properly.

Frankly, I was too focused on the Gear widget when I was doing this that I forgot about making sure the original widget was being handled properly. More in the next comment...